src/NuGet.Clients/PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -346,7 +346,7 @@ private void UpdatePackageList(IEnumerable<PackageItemListViewModel> packages, b
 
                     if (refresh)
                     {
-                        ClearPackageList();
+                        ClearPackageList(withLock: false);
```

Comment:
The only place it is called with lock is in `LoadItems#105`. I think it'd be better to remove locking from the `ClearPackageList` and use it in `LoadItems` instead (wrap the `ClearPackageList` call).